ckanext/resource_visibility/logic/action.py

```diff
@@ -39,6 +36,19 @@ def package_show(next_func, context, data_dict):
     return data_dict
 
 
+def remove_hidden_resources(context, data_dict):
+    for res in data_dict["resources"]:
+        if _is_resource_visible(res, data_dict):
+            continue
```

Comment:
Since this is the only action within the loop, why not just make it "if not visible then set flag"?

Reply:
It's more convenient for me.